ethereum/contracts/bridge/token/TokenState.sol

```diff
@@ -27,5 +29,54 @@ contract TokenStorage {
 }
 
 contract TokenState {
+    using Counters for Counters.Counter;
+
     TokenStorage.State _state;
+
+    // EIP712
+    // Cache the domain separator as an immutable value, but also store the chain id that it corresponds to, in order to
+    // invalidate the cached domain separator if the chain id changes.
+    string _version;
```

Comment:
we should consider moving all of these storage variables into `TokenStorage.State`. Generally it's a good idea for all storage slots to be defined in the same place to avoid future surprises

Reply:
Fixed. How does it look?